src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1197,11 +1201,24 @@ def join(
         # get the new mapped right on identifier
         right_on_cols = [right_identifiers_rename_map[key] for key in right_on_cols]
 
-        # Generate sql ON clause 'EQUAL_NULL(col1, col2) and EQUAL_NULL(col3, col4) ...'
         on = None
-        for left_col, right_col in zip(left_on_cols, right_on_cols):
-            eq = Column(left_col).equal_null(Column(right_col))
-            on = eq if on is None else on & eq
+
+        from snowflake.snowpark.modin.plugin._internal.join_utils import MatchComparator
+
+        # Use EQUAL_NULL as default to compare left and right "on" columns
+        on_comparators = (
```

Comment:
if we start support set of different operators, our optimization criteria at the very beginning will change for is_join_needed = True. only equal can be optimized